Source/Core/DolphinQt/Achievements/AchievementProgressWidget.cpp

```diff
@@ -51,6 +51,37 @@ AchievementProgressWidget::CreateAchievementBox(const rc_api_achievement_definit
 {
   if (!AchievementManager::GetInstance()->IsGameLoaded())
     return new QGroupBox();
+  QLabel* a_badge = new QLabel();
+  const auto unlock_status = AchievementManager::GetInstance()->GetUnlockStatus(achievement->id);
+  auto badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.locked_badge);
+  if (unlock_status.remote_unlock_status == AchievementManager::UnlockStatus::UnlockType::HARDCORE)
+  {
+    badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.unlocked_badge);
+  }
+  else if (hardcore_mode_enabled || unlock_status.session_unlock_count > 1)
+  {
+    badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.unlocked_badge);
+  }
+  else if (unlock_status.remote_unlock_status ==
+           AchievementManager::UnlockStatus::UnlockType::SOFTCORE)
+  {
+    badge = const_cast<AchievementManager::BadgeStatus*>(&unlock_status.unlocked_badge);
+  }
+  else if (unlock_status.session_unlock_count > 1)
```

Comment:
This statement will never be reached because you already check for this exact condition two statements earlier. Did you want an `&&` up there instead?